 drivers/media/rc/keymaps/Makefile  |  1 +
 drivers/media/rc/keymaps/rc-24k.c | 64 ++++++++++++++++++++++++++++++++++++++
 include/media/rc-map.h             |  1 +
 3 files changed, 54 insertions(+)
 create mode 100644 drivers/media/rc/keymaps/rc-24.c
 
 --- a/drivers/media/rc/keymaps/Makefile b/drivers/media/rc/keymaps/Makefile
+++ b/drivers/media/rc/keymaps/Makefile b/drivers/media/rc/keymaps/Makefile
@@ -1,4 +1,5 @@
-obj-$(CONFIG_RC_MAP) += rc-adstech-dvb-t-pci.o \
+obj-$(CONFIG_RC_MAP) += rc-24k.o \
+			rc-adstech-dvb-t-pci.o \
 			rc-alink-dtu-m.o \
 			rc-anysee.o \
 			rc-apac-viewcomp.o \

--- a/drivers/media/rc/keymaps/rc-24k.c
+++ b/drivers/media/rc/keymaps/rc-24k.c
@@ -0,0 +1,64 @@
+/* Keytable for RC-24K IR Remote Controller
+ *
+ * Copyright (c) 2017 Nagrace
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+#include <media/rc-map.h>
+#include <linux/module.h>
+
+static struct rc_map_table 24k[] = {
+	{ 0x404000, KEY_NUMERIC_0 },
+	{ 0x404001, KEY_NUMERIC_1 },
+	{ 0x404002, KEY_NUMERIC_2 },
+	{ 0x404003, KEY_NUMERIC_3 },
+	{ 0x404004, KEY_NUMERIC_4 },
+	{ 0x404005, KEY_NUMERIC_5 },
+	{ 0x404006, KEY_NUMERIC_6 },
+	{ 0x404007, KEY_NUMERIC_7 },
+	{ 0x404008, KEY_NUMERIC_8 },
+	{ 0x404009, KEY_NUMERIC_9 },
+	{ 0x404043, KEY_MUTE },
+	{ 0x40400b, KEY_UP },
+	{ 0x40400c, KEY_BACKSPACE },
+	{ 0x40400d, KEY_OK },
+	{ 0x40400e, KEY_DOWN },
+	{ 0x404010, KEY_LEFT },
+	{ 0x404011, KEY_RIGHT },
+	{ 0x404017, KEY_VOLUMEDOWN },
+	{ 0x404018, KEY_VOLUMEUP },
+	{ 0x40401a, KEY_HOME },
+	{ 0x404045, KEY_MENU },
+	{ 0x404042, KEY_BACK },
+	{ 0x404047, KEY_INFO },
+	{ 0x40404d, KEY_POWER },
+};
+
+static struct rc_map_list 24k_map = {
+	.map = {
+		.scan    = 24k,
+		.size    = ARRAY_SIZE(24k),
+		.rc_type = RC_TYPE_NEC,
+		.name    = RC_MAP_24K,
+	}
+};
+
+static int __init init_rc_map_24k(void)
+{
+	return rc_map_register(&24k_map);
+}
+
+static void __exit exit_rc_map_24k(void)
+{
+	rc_map_unregister(&24k_map);
+}
+
+module_init(init_rc_map_24k)
+module_exit(exit_rc_map_24k)
+
+MODULE_LICENSE("GPL");
+MODULE_AUTHOR("Nagrace");

--- a/include/media/rc-map.h
+++ b/include/media/rc-map.h
@@ -103,6 +103,7 @@
 
 /* Names of the several keytables defined in-kernel */
 
+#define RC_MAP_24K			  "rc-24k"
 #define RC_MAP_ADSTECH_DVB_T_PCI         "rc-adstech-dvb-t-pci"
 #define RC_MAP_ALINK_DTU_M               "rc-alink-dtu-m"
 #define RC_MAP_ANYSEE                    "rc-anysee"